packages/beacon-node/src/network/processor/aggregatorTracker.ts

```diff
@@ -16,8 +16,12 @@ export class AggregatorTracker {
 
   addAggregator(subnet: SubnetId, slot: Slot): void {
     this.subnetAggregatorsBySlot.getOrDefault(slot).add(subnet);
-
-    pruneSetToMax(this.subnetAggregatorsBySlot, MAX_SLOTS_CACHED);
+    pruneSetToMax(
+      this.subnetAggregatorsBySlot,
```

Comment:
The alternative could be to keep `subnetAggregatorsBySlot` sorted at insertion time? We might not have the necessary data structure yet.

Reply:
I had three options in mind
- sort keys when pruning (as implemented)
- switch to [`pruneBySlot`](https://github.com/ChainSafe/lodestar/blob/ae984f0f6079f78c2445f36ae2ac09c9e868197e/packages/beacon-node/src/chain/opPools/utils.ts#L10) logic, I am not sure if this strategy is correct here, although it could keep the cache size lower, maybe @tuyennhv has some thoughts
- as you suggested @jeluard always have keys sorted, I think we wanna keep a map for quick lookup times, although might not matter much since the array would have max 64 items

Reply:
Current approach is probably fine given the sizes involved, probably better not to over engineer yet :)

Reply:
I overlooked this yesterday, we have a ordered map

https://github.com/ChainSafe/lodestar/blob/ae984f0f6079f78c2445f36ae2ac09c9e868197e/packages/beacon-node/src/util/map.ts#L6

This one keeps track of the keys which is the tradeoff I guess

Reply:
Alternative using a ordered map https://github.com/ChainSafe/lodestar/compare/unstable...nflaig/ordered-map-aggregator-tracker

Reply:
turns out our "ordered" map isn't actually ordered... added units tests now

Reply:
> Current approach is probably fine given the sizes involved, probably better not to over engineer yet :)

I agree with this considering the cache size is just 64, may want to add a note for the reasoning we choose this approach